### PR TITLE
Deprecate MLIR CPP solver, step 1

### PR DESCRIPTION
**Pull request by jerryyin**
According to https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/141

Will completely remove the MIOpen cpp backend after june demo.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -321,20 +321,6 @@ if(MIOPEN_TEST_MLIR)
     set(IMPLICITGEMM_MLIR_ARGS_F ${IMPLICITGEMM_ARGS} --verbose --disable-backward-data --disable-backward-weights)
     set(IMPLICITGEMM_MLIR_ARGS_B ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-weights)
     set(IMPLICITGEMM_MLIR_ARGS_W ${IMPLICITGEMM_ARGS} --verbose --disable-forward --disable-backward-data)
-    
-    add_custom_test(test_conv_igemm_mlir ALLOW_NONXDLOPS
```
Comment (atamazov) [inline test/CMakeLists.txt:325]: Are these tests necessary for MLIR CPP backend only?

File: src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp [hunk 2]
```diff
@@ -57,46 +57,8 @@ ConvHipImplicitGemmMlirCppBwd::CalculateGemmSize(const ConvolutionContext& ctx)
 
 bool ConvHipImplicitGemmMlirCppBwd::IsApplicable(const ConvolutionContext& ctx) const
 {
-#if MIOPEN_USE_MLIR
-    if(miopen::IsDisabled(MIOPEN_DEBUG_CONV_HIP_IMPLICIT_GEMM_MLIR_CPP_BWD{}))
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp:61]: [Informative] Changing `miopen::IsDisabled(...)` to `!miopen::IsEnabled(...)` is the "standard" way for disabling a solver by default while keeping it available.

Comment (jerryyin) [reply src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp:61]: Will do! I agree this is a better way.

File: test/CMakeLists.txt (hunk 1 above)
Comment (jerryyin) [reply test/CMakeLists.txt:325]: The way I have it as now will make it run into merge conflict with #841. Once #841 is merged, I will rework this to get rid of cpp solver only, and keep bin solvers.

Comment (atamazov) [reply test/CMakeLists.txt:325]: Okay

File: src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp (hunk 2 above)
Comment (atamazov) [reply src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp:61]: Ok, I am expecting new commit(s) in this PR.

**Comment by atamazov** [comment]
@jerryyin Please resolve review comments & merge conflicts, thanks.

**Comment by jerryyin** [comment]
@atamazov Sorry I forgot to mention about it. I plan on to do update after #892. Between that PR and this PR is going to be conflicts again. It is a better time for me to remove cpp solver and do misc improvements al-together in this PR when all three directions of mlir binary backend in merged.

File: src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp (hunk 2 above)
Comment (jerryyin) [reply src/solver/conv_hip_implicit_gemm_mlir_cpp_bwd.cpp:61]: Done

**Review by atamazov — APPROVED** [review]

**Pull request merged**